tasks/ast_codegen/src/generators/derive_get_span.rs

```diff
@@ -0,0 +1,155 @@
+    }
+
+    fn generate(&mut self, ctx: &LateCtx) -> GeneratorOutput {
+        GeneratorOutput::Stream((
+            output(crate::AST_CRATE, "derive_get_span_mut.rs"),
+            derive::<true>(ctx),
+        ))
+    }
+}
+
+fn derive<const MUT: bool>(ctx: &LateCtx) -> TokenStream {
+    let (self_type, trait_ident, method_ident, result_type) = if MUT {
+        (
+            quote!(&mut self),
+            format_ident!("GetSpanMut"),
+            format_ident!("span_mut"),
+            quote!(&mut Span),
+        )
+    } else {
+        (quote!(&self), format_ident!("GetSpan"), format_ident!("span"), quote!(Span))
+    };
+
+    let derive_enum = |it: &EnumDef| {
+        let generics = it.generics();
+        let typ = it.to_type();
+        impl_trait(
+            &trait_ident,
+            &method_ident,
+            &generics,
+            &typ,
+            &self_type,
+            &result_type,
+            &derive_enum(it, &method_ident),
+        )
+    };
+
+    let derive_struct = |it: &StructDef| {
+        let generics = it.generics();
+        let typ = it.to_type();
+        impl_trait(
+            &trait_ident,
+            &method_ident,
+            &generics,
+            &typ,
+            &self_type,
+            &result_type,
+            &derive_struct::<MUT>(it, &method_ident),
+        )
+    };
+    let impls: Vec<TokenStream> = ctx
+        .schema
+        .definitions
+        .iter()
+        .filter(|def| def.visitable())
+        .map(|def| match &def {
+            TypeDef::Enum(it) => derive_enum(it),
+            TypeDef::Struct(it) => derive_struct(it),
+        })
+        .collect();
+
+    let header = generated_header!();
+
+    quote! {
+        #header
+        insert!("#![allow(clippy::match_same_arms)]");
+        endl!();
+
+        use crate::ast::*;
+        use oxc_span::{#trait_ident, Span};
+
+        #(#impls)*
+    }
+}
+
+fn derive_enum(def: &EnumDef, method: &Ident) -> TokenStream {
+    let matches = def.all_variants().map(|var| {
+        let ident = var.ident();
+        quote!(Self :: #ident(it) => it.#method())
+    });
+
+    quote! {
+        match self {
+            #(#matches),*
+        }
+    }
+}
+
+fn derive_struct<const MUT: bool>(def: &StructDef, method: &Ident) -> TokenStream {
+    let inner_span_hint = def.fields.iter().find(|it| it.markers.span);
+    if let Some(span_field) = inner_span_hint {
+        let ident = span_field.name.as_ref().map(ToIdent::to_ident).unwrap();
+        quote!(self.#ident.#method())
+    } else if MUT {
+        quote!(&mut self.span)
+    } else {
+        quote!(self.span)
+    }
+}
+
+fn impl_trait(
+    trait_ident: &Ident,
+    method_ident: &Ident,
+    generics: &Option<Generics>,
+    target_type: &Type,
+    self_: &TokenStream,
+    result_type: &TokenStream,
+    body: &TokenStream,
+) -> TokenStream {
+    quote! {
+        endl!();
+        impl #generics #trait_ident for #target_type {
+            #[inline]
```

Comment:
Previously `#[inline]` was only added for structs (if I remember right). I think it's unwise to put `#[inline]` on the enums as some of them are pretty large.

Reply:
Yeah, That was by mistake. Fixed it.

For ast builder I only inline enums with 5 or fewer variants, We can do something similar here as a follow-up.